plugin/storage/es/spanstore/schema.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package spanstore
+
+const spanMapping = `{
+   "settings":{
+      "index.mapping.nested_fields.limit":50,
+      "index.requests.cache.enable":true,
+      "index.mapper.dynamic":false,
+      "analysis":{
+         "analyzer":{
+            "traceId_analyzer":{
+               "type":"custom",
+               "tokenizer":"keyword",
+               "filter":"traceId_filter"
+            }
+         },
+         "filter":{
+            "traceId_filter":{
+               "type":"pattern_capture",
+               "patterns":[
+                  "([0-9a-f]{1,16})$"
+               ],
+               "preserve_original":true
+            }
+         }
+      }
+   },
+   "mappings":{
+      "_default_":{
+         "_all":{
+            "enabled":false
+         }
+      },
+      "span":{
+         "properties":{
+            "traceID":{
+               "type":"string",
+               "analyzer":"traceId_analyzer",
+               "fielddata":"true"
```

Comment:
My suggestion is to leave the concerns about the length of trace ID out of the ES schema. The collection pipeline can be configured to strip out the High portion of the traceID before saving, if there are some clients in the system that are still operating on 64bit IDs. I think 128 vs. 64 bit is a migration concern better left out of the storage implementation. In the storage the traceID is simply a blob of bytes, or a hex string, whichever is more efficient in the given storage (e.g. in Cassandra schema it's a byte blob).